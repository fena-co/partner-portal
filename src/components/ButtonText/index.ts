import styled, { css } from 'styled-components';

const ButtonContained = css`
  background: #13273f;
  color: #ffffff;
  border: none;
  cursor: pointer;
  &:hover {
    background-color: #19385e;
    filter: drop-shadow(0px 7px 20px rgba(108, 108, 138, 0.6));
  }
  &:disabled {
    background: #dbe3eb;
    filter: none;
    cursor: not-allowed;
  }
`;

const ButtonOutlined = css`
  background-color: #ffffff;
  border: 1px solid #c2cedb;
  color: #13273f;
  cursor: pointer;

  &:hover {
    box-sizing: border-box;
    border: 1px solid #13273f;
  }
  &:disabled {
    border: 1px solid #dbe3eb;
    box-sizing: border-box;
    color: #dbe3eb;
    cursor: not-allowed;
  }
`;

const ButtonText = styled.button<{ variant: string }>`
  font-family: Montserrat;
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;

  padding: 5px 15px;
  box-sizing: border-box;
  border-radius: 5px;

  ${({ variant }) =>
    variant === `contained` ? ButtonContained : ButtonOutlined}
`;

export default ButtonText;
