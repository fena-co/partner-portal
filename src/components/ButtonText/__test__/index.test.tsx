import ButtonText from '../index';
import renderer from 'react-test-renderer';

describe('Button Block', () => {
  it('renders correctly with variant contained ', () => {
    const tree = renderer
      .create(<ButtonText variant={`contained`}>Submit</ButtonText>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with variant outlined ', () => {
    const tree = renderer
      .create(<ButtonText variant={`outlined`}>Submit</ButtonText>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with variant outlined disabled', () => {
    const tree = renderer
      .create(<ButtonText variant={`outlined`}>Submit</ButtonText>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with variant contained disabled', () => {
    const tree = renderer
      .create(<ButtonText variant={`outlined`}>Submit</ButtonText>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
