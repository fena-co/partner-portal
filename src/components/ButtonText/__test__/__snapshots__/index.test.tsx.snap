// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Block renders correctly with variant contained  1`] = `
.c0 {
  font-family: Montserrat;
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
  padding: 5px 15px;
  box-sizing: border-box;
  border-radius: 5px;
  background: #13273f;
  color: #ffffff;
  border: none;
  cursor: pointer;
}

.c0:hover {
  background-color: #19385e;
  -webkit-filter: drop-shadow(0px 7px 20px rgba(108,108,138,0.6));
  filter: drop-shadow(0px 7px 20px rgba(108,108,138,0.6));
}

.c0:disabled {
  background: #dbe3eb;
  -webkit-filter: none;
  filter: none;
  cursor: not-allowed;
}

<button
  className="c0"
>
  Submit
</button>
`;

exports[`Button Block renders correctly with variant contained disabled 1`] = `
.c0 {
  font-family: Montserrat;
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
  padding: 5px 15px;
  box-sizing: border-box;
  border-radius: 5px;
  background-color: #ffffff;
  border: 1px solid #c2cedb;
  color: #13273f;
  cursor: pointer;
}

.c0:hover {
  box-sizing: border-box;
  border: 1px solid #13273f;
}

.c0:disabled {
  border: 1px solid #dbe3eb;
  box-sizing: border-box;
  color: #dbe3eb;
  cursor: not-allowed;
}

<button
  className="c0"
>
  Submit
</button>
`;

exports[`Button Block renders correctly with variant outlined  1`] = `
.c0 {
  font-family: Montserrat;
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
  padding: 5px 15px;
  box-sizing: border-box;
  border-radius: 5px;
  background-color: #ffffff;
  border: 1px solid #c2cedb;
  color: #13273f;
  cursor: pointer;
}

.c0:hover {
  box-sizing: border-box;
  border: 1px solid #13273f;
}

.c0:disabled {
  border: 1px solid #dbe3eb;
  box-sizing: border-box;
  color: #dbe3eb;
  cursor: not-allowed;
}

<button
  className="c0"
>
  Submit
</button>
`;

exports[`Button Block renders correctly with variant outlined disabled 1`] = `
.c0 {
  font-family: Montserrat;
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
  padding: 5px 15px;
  box-sizing: border-box;
  border-radius: 5px;
  background-color: #ffffff;
  border: 1px solid #c2cedb;
  color: #13273f;
  cursor: pointer;
}

.c0:hover {
  box-sizing: border-box;
  border: 1px solid #13273f;
}

.c0:disabled {
  border: 1px solid #dbe3eb;
  box-sizing: border-box;
  color: #dbe3eb;
  cursor: not-allowed;
}

<button
  className="c0"
>
  Submit
</button>
`;
